pkg/notification/notification.go

```diff
@@ -24,8 +24,14 @@ type Event struct {
 	Timestamp time.Time
 }
 
-type EventSource struct {
-	Events        <-chan Event
+type EventSource interface {
```

Comment:
do we need this interface?

Reply:
I have used this interface so the UnlimitedEventSource and the RateLimited one could be handled through it without knowing which one you are using. In fact RateLimitedEventSource just decorates another EventSource adding the rate limitation to it.

Reply:
Sure, but  do we need an explicit declaration? IOW, do we need this level of abstraction yet or can the builder function just return concrete objects?

Reply:
builder function creates different type of event source from different arguments and should be declared to return only one type so I think the interface is needed, at least to the best of my knowledge.

Reply:
ok, if nothing else this interface enabled better testing (enables a form of dependency injection) so let's keep it!